local-cli/runAndroid/runAndroid.js

```diff
@@ -78,13 +78,108 @@ function tryRunAdbReverse(device) {
 // Builds the app and runs it on a connected emulator / device.
 function buildAndRun(args) {
   process.chdir(path.join(args.root, 'android'));
+  const cmd = process.platform.startsWith('win')
+    ? 'gradlew.bat'
+    : './gradlew';
+
+  const packageName = fs.readFileSync(
+      'app/src/main/AndroidManifest.xml',
+      'utf8'
+    ).match(/package="(.+?)"/)[1];
+
+  const adbPath = getAdbPath();
+  if (args.deviceId) {
+    runOnSpecificDevice(args, cmd, packageName, adbPath);
+  } else {
+    runOnAllDevices(args, cmd, packageName, adbPath);
+  }
+}
+
+function runOnSpecificDevice(args, cmd, packageName, adbPath) {
+  let devices = adb.getDevices();
+  let selectedDevice = findMatchingDevice(devices, args.deviceId);
+  if (selectedDevice){
+    buildApk(cmd);
+    installAndLaunchOnDevice(args, selectedDevice, packageName, adbPath);
+  } else {
+      if (devices){
+        console.log('Could not find device with the id: "' + args.deviceId + '".');
+        console.log('Choose one of the following:');
+        child_process.execFileSync('adb', ['devices'], {
+          stdio: [process.stdin, process.stdout, process.stderr],
+        });
+      } else {
+        console.log('No android devices connected.');
+      }
+  }
+}
+
+function buildApk(cmd) {
   try {
-    adb.getDevices().map((device) => tryRunAdbReverse(device));
+    console.log(chalk.bold(
+      'Building the app...'
+    ));
+
+    child_process.execFileSync(cmd, ['build', '-x', 'lint'], {
```

Comment:
Maybe we should call the `cmd` `gradlew` - also in the function calls that call `buildApk`. This way it's more obvious it's always `gradlew` or `gradle.bat` command - it is, right?

At fist I got confused when I saw `build -x lint` being appended to a command and didn't know what the command could be.

What does `-x lint` do? Can you add a comment above this line explaining why use `-x lint`?